homeassistant/components/intellidrive/entity.py

```diff
@@ -0,0 +1,56 @@
+"""Entity for the opengarage.io component."""
+from __future__ import annotations
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, STATUSDICT_SERIALNO, STATUSDICT_VERSION
+from .coordinator import ReisingerCoordinator
+
+
+class IntelliDriveEntity(CoordinatorEntity[ReisingerCoordinator]):
+    """Representation of a IntelliDrive entity, which is used as a base entity for future entities."""
+
+    def __init__(
+        self,
+        coordinator: ReisingerCoordinator,
+        device_id: str,
+        description: EntityDescription | None = None,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+
+        if description is not None:
+            self.entity_description = description
+            self._attr_unique_id = f"{device_id}_{description.key}"
+        else:
+            self._attr_unique_id = device_id
+
+        self._device_id = device_id
+        self._update_attr()
+
+    @callback
+    def _update_attr(self) -> None:
+        """Update the state and attributes."""
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attr()
+        self.async_write_ha_state()
```

Comment:
Not sure, but I think it is typically better to do native_value, instead of handle_coordinator_update

Reply:
I am also not sure, but I cannot find an overrideable method or attribute thats called "native_value" in the CoordinatorEntity or BaseCoordinatorEntity, or how can I use your solution?
Thanks for your feedback 👍